apiserver/facades/agent/provisioner/provisioninginfo.go

```diff
@@ -302,6 +315,39 @@ func (p *ProvisionerAPI) machineSubnetsAndZones(m *state.Machine) (map[string][]
 	return subnetsToZones, nil
 }
 
+func (p *ProvisionerAPI) machineLXDProfiles(m *state.Machine, env environs.Environ) ([]string, error) {
+	profileEnv, ok := env.(lxd.LXDProfiler)
+	if !ok {
+		return nil, nil
+	}
+	units, err := m.Units()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	names := make([]string, 0)
+	for _, unit := range units {
+		app, err := unit.Application()
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		ch, _, err := app.Charm()
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		profile := ch.LXDProfile()
+		if profile == nil || (profile != nil && profile.Empty()) {
+			continue
+		}
+		pName := fmt.Sprintf("juju-%s-%s-%d-%d", profileEnv.Name(), app.Name(), ch.Revision(), unit.UnitTag().Number())
```

Comment:
I think we can drop `profileEnv.Name()` here, as the `lxd-profile.yaml` will be the same for name and revision, even if it's on a different provider. They'll be on differing models if you do want to identify between the different providers.

Reply:
So Name() gives us the model name. Unless there's a different way in the provisioner code to get it.

Reply:
A machine can only appear in one model. I'm confused why we're concerned with cross model differentiation.

Reply:
@mitechie, lxd doesn't know about models.  once we ask lxd to create a profile, it's name must be such that juju can determine which model the profile should be used in.